assets/src/edit-story/components/canvas/useUploadWithPreview.js

```diff
@@ -53,28 +64,37 @@ function useUploadWithPreview() {
         'lengthFormatted',
         'id',
       ]);
-      const updatedResource = {
-        ...element.resource,
-        ...keysToUpdate,
-      };
-      updateElementById({
-        elementId: element.id,
-        properties: {
-          resource: updatedResource,
+      updateElementsByResourceId({
+        // We want to update all resources without id (still uploading)
+        id: undefined,
+        properties: (el) => {
+          if (el.resource?.src === blobUrl) {
+            const updatedResource = {
+              ...el.resource,
+              ...keysToUpdate,
+            };
+            return {
+              resource: updatedResource,
+            };
+          }
+          return {};
         },
       });
       if (resource.type === 'video') {
         await uploadVideoPoster(resource.id, resource.src);
       }
     },
-    [updateElementById, uploadVideoPoster]
+    [updateElementsByResourceId, uploadVideoPoster]
   );
 
   const onUploadFailure = useCallback(
     ({ element }) => {
       deleteElementById({ elementId: element.id });
+      showSnackbar({
+        message: __('Upload failed, preview was removed', 'web-stories'),
```

Comment:
Perhaps "the element was removed" would be better here? Not sure the user is aware at all that it's just the preview shown initially, for the user it likely seems the "real thing". Or is this from Figma?

Reply:
I think there is only this in Figma: 
![image](https://user-images.githubusercontent.com/21036224/85797411-d869fe80-b73b-11ea-9180-bb071b91cc45.png)

Will use the "element" for now. It will probably change after we agree on the failure case.